### PR TITLE
Fix warnings and android issues

### PR DESCRIPTION
**Pull request by DreamNik**
Here's fixes for building for Android platform:
1) Fix compilation errors and warnings.
2) Changes related to building for Android using clang, Qt6 and fresh 3rdparty libraries, but could be useful when building in other environments too.
3) Fix plugin loading on android.

### CODE REVIEW

**Comment by f4exb** [comment]
Sorry this breaks Metis MISO I am going to revert this PR

File: plugins/samplemimo/metismiso/metismisoudphandler.cpp [hunk 1]
```diff
@@ -622,8 +622,6 @@ void MetisMISOUDPHandler::processIQBuffer(unsigned char* buffer)
                 }
             }
 
-            sampleMic  = (int)((signed char) buffer[b++]) << 8;
-            sampleMic += (int)((unsigned char)buffer[b++]);
```
Comment (f4exb) [inline plugins/samplemimo/metismiso/metismisoudphandler.cpp:626]: This is a breaking change and I don't see the point of it.

Comment (DreamNik) [reply plugins/samplemimo/metismiso/metismisoudphandler.cpp:626]: The variable "sampleMic" isn't used anywhere, so compiler complained about it, and thus I've removed it.

Comment (srcejon) [reply plugins/samplemimo/metismiso/metismisoudphandler.cpp:626]: b is modified and used though.

Comment (DreamNik) [reply plugins/samplemimo/metismiso/metismisoudphandler.cpp:626]: Yes, you're right! Didn't noticed that.

**Pull request merged**